lib/internal/modules/esm/loader.js

```diff
@@ -78,6 +78,8 @@ class Loader {
   }
 
   async resolve(specifier, parentURL) {
+    if (parentURL instanceof URL)
```

Comment:
is there a more robust test than `instanceof` here? the `vm` module could produce a URL that's not `===` to this realm's `URL`.

It might be simpler to just use `.href` in the tests, and delay this semver-minor change to a separate PR?

Reply:
I had a look at what the fs functions do and adopted the same approach in using the `searchParams` symbol as the check as in https://github.com/nodejs/node/blob/master/lib/internal/url.js.

The alternative is to use the `toString()` method by default instead of strict argument checking. I'm tempted to say that might be more in line with web behaviour (eg `window.location = { toString () { return 'https://www.google.com' } }`). I also know URL as an API argument is discouraged in the URL spec (despite this being something everyone does!).

Open to feedback either way here further, or splitting it out if that is deemed better. I'm just trying to hit my PR budget for this month :P

Reply:
1. totally agree that it should match what the other things in core do; changing them should be its own PR
1. using toString is reasonable but probably will cause more bugs. what would be ideal is imo if URL had a cross-realm brand check available, and then all `instanceof URL` uses in core would change to use that.
1. it still seems to me like it'd be better to not make an unrelated production code change just to make tests slightly more convenient :-) ie, i'd still prefer splitting this into a separate PR - but it's your/collabs' call, ofc!

Reply:
Agreed. Yeah it seems like it's not clear enough to be a simple change - ok I've reverted the URL change for now.